lantiq: fix boot isues on danube boards

abajk opened the pull request:
Enabling SMP on Danube[1] is incompatible with a patch that adds support for interrupt handling on all cores on other  platforms[2]. This patch fixes the mentioned issue.

1. 084c20f6c54180f2dc6b3efc1c5ba9b62afbaf26 ("lantiq: xway: kernel: enable SMP support ")
2. fbd33d61648ae8982fbada7ad3b6d8222b367ab5 ("lantiq: enable interrupts on second VPEs")

Fixes: #13934
Fixes: #14283
Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>

Leo-PL commented:
Yes, please. Stumbled upon this on Easybox 802.

Leo-PL commented:
ping @hauke @Ansuel

Ansuel commented:
@Leo-PL ugh... Would be lovely to find the cause tho instead of reverting :( Since this is present only on some device. (is this correct?)

Leo-PL commented:
> @Leo-PL ugh... Would be lovely to find the cause tho instead of reverting :( Since this is present only on some device. (is this correct?)

We may continue to investigate that on development branches, but I believe on stable (23.05) it should be reverted, because this causes devices to brick, and at least mine - required painful disassembly to fix.

Ansuel commented:
@Leo-PL yes I will cherry-pick this and merge on openwrt-23.05
Do we have a clear list of the affected targets so we can add them to the known issue list?

Leo-PL commented:
From what I managed to compile by quickly looking over the linked issues, this seems limited to Arcadyan boxes, but of several models:
ARV752DPW
ARV752DPW22
ARV7506PW11
ARV8539PW22

This list is very likely incomplete.

abajk commented:
@Leo-PL @Ansuel 

It seems that none of the devices based on the danube chipset (single core MIPS 24Kc) should boot with SMP enabled. Booting hangs in the plat_smp_setup() function because the struct smp_ops is not initialized[1].

[1] https://elixir.bootlin.com/linux/v5.15.147/source/arch/mips/include/asm/smp-ops.h#L48

Ansuel commented:
@abajk can you test a simple patch for this?

abajk commented:
> @abajk can you test a simple patch for this?

@Ansuel  yes

FYI. I have tested a simple fix:
~~~
@@ -45,7 +45,12 @@ static inline void plat_smp_setup(void)
 {
        extern const struct plat_smp_ops *mp_ops;       /* private */
 
-       mp_ops->smp_setup();
+       printk(KERN_INFO "%s:1 smp_ops=%p\n", __func__, mp_ops);
+       if (mp_ops != NULL) {
+               printk(KERN_INFO "%s:1 smp_setup=%p\n", __func__, mp_ops->smp_setup);
+               mp_ops->smp_setup();
+       }
+       printk(KERN_INFO "%s:2\n", __func__);
~~~

Booting crashes later:

~~~
[    0.000000] Linux version 5.15.147 (aleksander@laptop-olek) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r24879-7812b2848f) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 SMP Sun Jan 21 15:53:55 2024
[    0.000000] SoC: Danube rev 1.5
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019641 (MIPS 24KEc)
[    0.000000] MIPS: machine is Arcor 802
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] plat_smp_setup:1 smp_ops=00000000
[    0.000000] plat_smp_setup:2
[    0.000000] Primary instruction cache 16kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 16kB, 4-way, VIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] percpu: Embedded 10 pages/cpu s12016 r8192 d20752 u40960

~~~
~~~

Ansuel commented:
@abajk it's an interesting finding for sure... I can expect other case like that... maybe we should check the difference with SMP on and off? Is there a lot in the MIPS code?

abajk committed:
lantiq: fix boot isues on danube boards

Enabling SMP on Danube[1] is incompatible with a patch that
adds support for interrupt handling on all cores on other
platforms[2]. This patch fixes the mentioned issue.

1. 084c20f6c54180f2dc6b3efc1c5ba9b62afbaf26 ("lantiq: xway: kernel: enable SMP support ")
2. fbd33d61648ae8982fbada7ad3b6d8222b367ab5 ("lantiq: enable interrupts on second VPEs")

Fixes: #13934
Fixes: #14283
Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>

abajk commented:
> @abajk it's an interesting finding for sure... I can expect other case like that... maybe we should check the difference with SMP on and off? Is there a lot in the MIPS code?

I have prepared the second iteration of the fix. At the moment I tested it only on Easybox 802 (Lantiq Danube).

Ansuel commented:
@abajk patch looks good and ready for upstreaming but does it fix boot or it's still stuck after?

abajk commented:
> @abajk patch looks good and ready for upstreaming but does it fix boot or it's still stuck after?

@Ansuel The kernel boots correctly and I can log in via LUCI.

As far as I can see only these logs are suspicious:
~~~
...
[    2.040406] number of CFI chips: 1
[    2.043786] 4 fixed-partitions partitions found on MTD device ltq_nor
[    2.049960] Creating 4 MTD partitions on "ltq_nor":
[    2.054584] 0x000000000000-0x000000010000 : "uboot"
[    2.064322] 0x000000010000-0x000000020000 : "u-boot-env"
[    2.071597] 0x000000020000-0x0000007f0000 : "firmware"
[    2.079123] 2 uimage-fw partitions found on MTD device firmware
[    2.083766] 0x000000020000-0x0000002976c3 : "kernel"
[    2.088560] mtd: partition "kernel" doesn't end on an erase/write block -- force read-only
[    2.100149] 0x0000002976c3-0x0000007f0000 : "rootfs"
[    2.103599] mtd: partition "rootfs" doesn't start on an erase/write block boundary -- force read-only
[    2.116023] mtd: setting mtd4 (rootfs) as root device
[    2.119834] 1 squashfs-split partitions found on MTD device rootfs
[    2.125567] 0x0000006e0000-0x0000007f0000 : "rootfs_data"
[    2.134238] 0x0000007f0000-0x000000800000 : "board_config"
[   18.225491] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.232697] epc = 77dc5b34 in libc.so[77d52000+ab000]
[   18.237398] ra  = 004020b9 in kmodloader[400000+3000]
[   18.293242] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.300441] epc = 77e0db34 in libc.so[77d9a000+ab000]
[   18.305143] ra  = 004020b9 in kmodloader[400000+3000]
[   18.359957] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.366967] epc = 77e48b34 in libc.so[77dd5000+ab000]
[   18.372058] ra  = 004020b9 in kmodloader[400000+3000]
[   18.425944] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.433151] epc = 77e66b34 in libc.so[77df3000+ab000]
[   18.437844] ra  = 004020b9 in kmodloader[400000+3000]
[   18.492347] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.499551] epc = 77e05b34 in libc.so[77d92000+ab000]
[   18.504254] ra  = 004020b9 in kmodloader[400000+3000]
[   18.559798] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.566804] epc = 77e99b34 in libc.so[77e26000+ab000]
[   18.571896] ra  = 004020b9 in kmodloader[400000+3000]
[   18.625522] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.632732] epc = 77e73b34 in libc.so[77e00000+ab000]
[   18.637428] ra  = 004020b9 in kmodloader[400000+3000]
[   18.677327] Registering RTL8306SDM switch with Chip ID: 0x5988, version: 0x0000
[   18.683238] Realtek RTL8306S 1e180000.etop-ffffffff:00: attached PHY driver (mii_bus:phy_addr=1e180000.etop-ffffffff:00, irq=POLL)
[   18.708335] NET: Registered PF_INET6 protocol family
[   18.751447] Segment Routing with IPv6
[   18.753847] In-situ OAM (IOAM) with IPv6
[   18.757774] NET: Registered PF_PACKET protocol family
[   18.762767] 8021q: 802.1Q VLAN Support v1.8
[   18.818552] Freeing unused kernel image (initmem) memory: 5376K
[   18.823079] This architecture does not have kernel memory protection.
[   18.829309] Run /init as init process
[   20.998853] init: Console is alive
...
~~~

Ansuel commented:
@abajk very nice work! Do we have bootlog with SMP disabled so we can make some comparison? Seems something wrong is happening in the memory (maybe related?)

abajk commented:
> @abajk very nice work! Do we have bootlog with SMP disabled so we can make some comparison? Seems something wrong is happening in the memory (maybe related?)

@Ansuel 

SMP Disabled:
~~~
Starting kernel ...

[    0.000000] Linux version 5.15.147 (aleksander@laptop-olek) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r24879-7812b2848f) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 Sun Jan 21 13:13:06 2024
[    0.000000] SoC: Danube rev 1.5
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019641 (MIPS 24KEc)
[    0.000000] MIPS: machine is Arcor 802
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 16kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 16kB, 4-way, VIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16240
[    0.000000] Kernel command line: console=ttyLTQ0,115200
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes, linear)
[    0.000000] Writing ErrCtl register=0001d9f0
[    0.000000] Readback ErrCtl register=0001d9f0
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 55148K/65536K available (6323K kernel code, 604K rwdata, 1356K rodata, 1172K init, 213K bss, 10388K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 256
[    0.000000] CPU Clock: 333MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 11467562657 ns
[    0.000004] sched_clock: 32 bits at 166MHz, resolution 6ns, wraps every 12884901885ns
[    0.008128] Calibrating delay loop... 221.18 BogoMIPS (lpj=1105920)
[    0.094314] pid_max: default: 32768 minimum: 301
[    0.101763] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.108980] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.139663] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.149464] futex hash table entries: 256 (order: -1, 3072 bytes, linear)
[    0.156758] pinctrl core: initialized pinctrl subsystem
[    0.166190] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.199349] pinctrl-xway 1e100b10.pinmux: Init done
[    0.214248] dma-xway 1e104100.dma: Init done - hw rev: 3, ports: 5, channels: 20
[    0.261132] usbcore: registered new interface driver usbfs
[    0.266777] usbcore: registered new interface driver hub
[    0.272119] usbcore: registered new device driver usb
[    0.280083] PCI host bridge to bus 0000:00
[    0.284051] pci_bus 0000:00: root bus resource [mem 0x18000000-0x19ffffff]
[    0.291072] pci_bus 0000:00: root bus resource [io  0x1ae00000-0x1affffff]
[    0.297935] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.309384] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    0.316999] clocksource: Switched to clocksource MIPS
[    0.326299] NET: Registered PF_INET protocol family
[    0.331664] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.340580] tcp_listen_portaddr_hash hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.349064] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.356759] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.364567] TCP bind hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.371674] TCP: Hash tables configured (established 1024 bind 1024)
[    0.378423] UDP hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.384877] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.393015] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.398761] PCI: CLS 0 bytes, default 32
[    0.402655] gptu: totally 6 16-bit timers/counters
[    0.407702] gptu: misc_register on minor 127
[    0.411884] gptu: succeeded to request irq 126
[    0.416384] gptu: succeeded to request irq 127
[    0.421005] gptu: succeeded to request irq 128
[    0.425522] gptu: succeeded to request irq 129
[    0.430030] gptu: succeeded to request irq 130
[    0.434439] gptu: succeeded to request irq 131
[    0.445705] lantiq,vmmc 1f107000.vmmc: requested GPIO 465
[    0.451097] lantiq,vmmc 1f107000.vmmc: reserved 1MB at 0x(ptrval)
[    0.469945] workingset: timestamp_bits=14 max_order=14 bucket_order=0
[    0.494765] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.500587] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.519262] 1e100c00.serial: ttyLTQ0 at MMIO 0x1e100c00 (irq = 112, base_baud = 0) is a lantiq,asc
[    0.528263] printk: console [ttyLTQ0] enabled
[    0.528263] printk: console [ttyLTQ0] enabled
[    0.536639] printk: bootconsole [early0] disabled
[    0.536639] printk: bootconsole [early0] disabled
[    0.548700] lantiq nor flash device: 00800000 at 10000000
[    0.552889] ltq_nor: Found 1 x16 devices at 0x0 in 16-bit bank. Manufacturer ID 0x0000c2 Chip ID 0x0022cb
[    0.561984] Amd/Fujitsu Extended Query Table at 0x0040
[    0.566893]   Amd/Fujitsu Extended Query version 1.1.
[    0.571826] number of CFI chips: 1
[    0.575224] 4 fixed-partitions partitions found on MTD device ltq_nor
[    0.581380] Creating 4 MTD partitions on "ltq_nor":
[    0.586121] 0x000000000000-0x000000010000 : "uboot"
[    0.607944] 0x000000010000-0x000000020000 : "u-boot-env"
[    0.614417] 0x000000020000-0x0000007f0000 : "firmware"
[    0.624401] 2 uimage-fw partitions found on MTD device firmware
[    0.628880] 0x000000020000-0x0000002976c3 : "kernel"
[    0.633590] mtd: partition "kernel" doesn't end on an erase/write block -- force read-only
[    0.644267] 0x0000002976c3-0x0000007f0000 : "rootfs"
[    0.647809] mtd: partition "rootfs" doesn't start on an erase/write block boundary -- force read-only
[    0.662312] mtd: setting mtd4 (rootfs) as root device
[    0.665948] 1 squashfs-split partitions found on MTD device rootfs
[    0.671952] 0x0000006e0000-0x0000007f0000 : "rootfs_data"
[    0.679862] 0x0000007f0000-0x000000800000 : "board_config"
[    0.796128] Registering RTL8306SDM switch with Chip ID: 0x5988, version: 0x0000
[    0.802001] Realtek RTL8306S 1e180000.etop-ffffffff:00: attached PHY driver (mii_bus:phy_addr=1e180000.etop-ffffffff:00, irq=POLL)
[    0.826525] NET: Registered PF_INET6 protocol family
[    0.862440] Segment Routing with IPv6
[    0.864906] In-situ OAM (IOAM) with IPv6
[    0.868842] NET: Registered PF_PACKET protocol family
[    0.873468] 8021q: 802.1Q VLAN Support v1.8
[    0.917113] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    0.929949] Freeing unused kernel image (initmem) memory: 1172K
[    0.934356] This architecture does not have kernel memory protection.
[    0.940688] Run /sbin/init as init process
[    2.636027] init: Console is alive
[    2.638952] init: - watchdog -
[    6.032351] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    6.205232] dwc2 1e101000.usb: supply vusb_d not found, using dummy regulator
[    6.211402] dwc2 1e101000.usb: supply vusb_a not found, using dummy regulator
[    6.261146] dwc2 1e101000.usb: DWC OTG Controller
[    6.264433] dwc2 1e101000.usb: new USB bus registered, assigned bus number 1
[    6.271355] dwc2 1e101000.usb: irq 62, io mem 0x1e101000
[    6.279202] hub 1-0:1.0: USB hub found
[    6.283099] hub 1-0:1.0: 1 port detected
[    6.295770] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    6.312509] init: - preinit -
[    6.411211] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[    6.416902] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[    6.423982] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[    7.236666] usb 1-1: new high-speed USB device number 2 using dwc2
[    7.241651] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[    7.248431] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[    7.255355] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[    7.914427] hub 1-1:1.0: USB hub found
[    7.916903] hub 1-1:1.0: 4 ports detected
[    8.143661] random: procd: uninitialized urandom read (4 bytes read)
[    9.241722] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    9.527157] random: crng init done
[   12.004699] jffs2: notice: (267) jffs2_build_xattr_subsystem: complete building xattr subsystem, 6 of xdatum (0 unchecked, 1 orphan) and 7 of xref (1 dead, 0 orphan) found.
[   12.023194] mount_root: switching to jffs2 overlay
[   12.038194] overlayfs: upper fs does not support tmpfile.
[   12.066428] urandom-seed: Seeding with /etc/urandom.seed
[   12.484932] procd: - early -
[   12.487334] procd: - watchdog -
[   13.584652] procd: - watchdog -
[   13.588743] procd: - ubus -
[   13.996724] procd: - init -
Please press Enter to activate this console.
[   17.938032] kmodloader: loading kernel modules from /etc/modules.d/*
[   19.064663] IFXOS, Version 1.7.1 (c) Copyright 2009, Lantiq Deutschland GmbH
[   19.114061] Lantiq TAPI device driver, version 3.13.0.2, (c) 2001-2010 Lantiq Deutschland GmbH
[   19.146503] NET: Registered PF_ATMPVC protocol family
[   19.150157] NET: Registered PF_ATMSVC protocol family
[   19.193853] PPP generic driver version 2.4.2
[   19.225939] IFX MEI Version 5.00.00
[   19.288185] Infineon CPE API Driver version: DSL CPE API V3.24.4.4
[   19.347612] Loading modules backported from Linux version v6.5-0-g2dde18cd1d8f
[   19.353303] Backport generated by backports.git v5.15.92-1-56-g5fb2ccb6b9e8
[   19.406149] Lantiq MIPS24KEc MPS driver, version 2.2.2.5, (c) 2006-2010 Lantiq Deutschland GmbH
[   19.413407] request_timer(3, 0x000001AE, 1)...
[   19.413499] successful!
[   19.420567] Lantiq VMMC device driver, version 1.9.0.3, (c) 2006-2010 Lantiq Deutschland GmbH
[   19.524006] NET: Registered PF_PPPOX protocol family
[   21.098008] kmodloader: done loading kernel modules from /etc/modules.d/*
[   26.298212] urngd: v1.0.2 started.



BusyBox v1.36.1 (2024-01-21 13:13:06 UTC) built-in shell (ash)


  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt SNAPSHOT, r24879-7812b2848f
 -----------------------------------------------------
=== WARNING! =====================================
There is no root password defined on this device!
Use the "passwd" command to set up a new password
in order to prevent unauthorized SSH logins.
--------------------------------------------------
root@OpenWrt:/# 
root@OpenWrt:/# [   80.497402] leds red:umts: Failed to get trigger sources for /leds/umts
[   85.647763] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   85.690402] br-lan: port 1(eth0.1) entered blocking state
[   85.694322] br-lan: port 1(eth0.1) entered disabled state
[   85.700709] device eth0.1 entered promiscuous mode
[   85.704185] device eth0 entered promiscuous mode
[   85.767366] br-lan: port 1(eth0.1) entered blocking state
[   85.771302] br-lan: port 1(eth0.1) entered forwarding state
[   86.668077] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
~~~

SMP Enabled:
~~~
Starting kernel ...

[    0.000000] Linux version 5.15.147 (aleksander@laptop-olek) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r24879-7812b2848f) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 SMP Sun Jan 21 18:18:06 2024
[    0.000000] SoC: Danube rev 1.5
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019641 (MIPS 24KEc)
[    0.000000] MIPS: machine is Arcor 802
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 16kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 16kB, 4-way, VIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] percpu: Embedded 10 pages/cpu s12016 r8192 d20752 u40960
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16240
[    0.000000] Kernel command line: console=ttyLTQ0,115200
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes, linear)
[    0.000000] Writing ErrCtl register=0001d9f0
[    0.000000] Readback ErrCtl register=0001d9f0
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 50448K/65536K available (6740K kernel code, 619K rwdata, 1392K rodata, 5376K init, 216K bss, 15088K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=2 to nr_cpu_ids=1.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=1
[    0.000000] NR_IRQS: 256
[    0.000000] CPU Clock: 333MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 11467562657 ns
[    0.000003] sched_clock: 32 bits at 166MHz, resolution 6ns, wraps every 12884901885ns
[    0.008182] Calibrating delay loop... 221.18 BogoMIPS (lpj=1105920)
[    0.094464] pid_max: default: 32768 minimum: 301
[    0.101995] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.109293] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.128764] rcu: Hierarchical SRCU implementation.
[    0.135226] smp: Bringing up secondary CPUs ...
[    0.139627] smp: Brought up 1 node, 1 CPU
[    0.157387] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.167228] futex hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.174394] pinctrl core: initialized pinctrl subsystem
[    0.184064] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.218833] pinctrl-xway 1e100b10.pinmux: Init done
[    0.233794] dma-xway 1e104100.dma: Init done - hw rev: 3, ports: 5, channels: 20
[    0.292217] usbcore: registered new interface driver usbfs
[    0.298085] usbcore: registered new interface driver hub
[    0.303426] usbcore: registered new device driver usb
[    0.311642] PCI host bridge to bus 0000:00
[    0.315615] pci_bus 0000:00: root bus resource [mem 0x18000000-0x19ffffff]
[    0.322663] pci_bus 0000:00: root bus resource [io  0x1ae00000-0x1affffff]
[    0.329496] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.341085] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    0.349478] clocksource: Switched to clocksource MIPS
[    0.358817] NET: Registered PF_INET protocol family
[    0.364234] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.373472] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes, linear)
[    0.381881] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.389681] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.397387] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.404546] TCP: Hash tables configured (established 1024 bind 1024)
[    0.411270] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.417758] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.425870] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.431728] PCI: CLS 0 bytes, default 32
[    0.435670] gptu: totally 6 16-bit timers/counters
[    0.440705] gptu: misc_register on minor 127
[    0.444944] gptu: succeeded to request irq 126
[    0.449688] gptu: succeeded to request irq 127
[    0.454198] gptu: succeeded to request irq 128
[    0.458607] gptu: succeeded to request irq 129
[    0.463248] gptu: succeeded to request irq 130
[    0.467755] gptu: succeeded to request irq 131
[    0.478906] lantiq,vmmc 1f107000.vmmc: requested GPIO 465
[    0.484333] lantiq,vmmc 1f107000.vmmc: reserved 1MB at 0x(ptrval)
[    1.933112] workingset: timestamp_bits=14 max_order=14 bucket_order=0
[    1.956411] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    1.962268] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    1.981881] 1e100c00.serial: ttyLTQ0 at MMIO 0x1e100c00 (irq = 112, base_baud = 0) is a lantiq,asc
[    1.990934] printk: console [ttyLTQ0] enabled
[    1.990934] printk: console [ttyLTQ0] enabled
[    1.999280] printk: bootconsole [early0] disabled
[    1.999280] printk: bootconsole [early0] disabled
[    2.017079] lantiq nor flash device: 00800000 at 10000000
[    2.021451] ltq_nor: Found 1 x16 devices at 0x0 in 16-bit bank. Manufacturer ID 0x0000c2 Chip ID 0x0022cb
[    2.030400] Amd/Fujitsu Extended Query Table at 0x0040
[    2.035282]   Amd/Fujitsu Extended Query version 1.1.
[    2.040239] number of CFI chips: 1
[    2.043619] 4 fixed-partitions partitions found on MTD device ltq_nor
[    2.049793] Creating 4 MTD partitions on "ltq_nor":
[    2.054417] 0x000000000000-0x000000010000 : "uboot"
[    2.063642] 0x000000010000-0x000000020000 : "u-boot-env"
[    2.070969] 0x000000020000-0x0000007f0000 : "firmware"
[    2.078544] 2 uimage-fw partitions found on MTD device firmware
[    2.083171] 0x000000020000-0x0000002976c3 : "kernel"
[    2.087742] mtd: partition "kernel" doesn't end on an erase/write block -- force read-only
[    2.099208] 0x0000002976c3-0x0000007f0000 : "rootfs"
[    2.102841] mtd: partition "rootfs" doesn't start on an erase/write block boundary -- force read-only
[    2.115173] mtd: setting mtd4 (rootfs) as root device
[    2.118822] 1 squashfs-split partitions found on MTD device rootfs
[    2.124879] 0x0000006e0000-0x0000007f0000 : "rootfs_data"
[    2.133236] 0x0000007f0000-0x000000800000 : "board_config"
[   18.205794] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.212988] epc = 77e82b34 in libc.so[77e0f000+ab000]
[   18.217699] ra  = 004020b9 in kmodloader[400000+3000]
[   18.273549] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.280749] epc = 77db9b34 in libc.so[77d46000+ab000]
[   18.285452] ra  = 004020b9 in kmodloader[400000+3000]
[   18.340164] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.347167] epc = 77dcfb34 in libc.so[77d5c000+ab000]
[   18.352259] ra  = 004020b9 in kmodloader[400000+3000]
[   18.406167] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.413373] epc = 77de8b34 in libc.so[77d75000+ab000]
[   18.418068] ra  = 004020b9 in kmodloader[400000+3000]
[   18.472664] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.479859] epc = 77e75b34 in libc.so[77e02000+ab000]
[   18.484563] ra  = 004020b9 in kmodloader[400000+3000]
[   18.540550] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.547553] epc = 77e62b34 in libc.so[77def000+ab000]
[   18.552648] ra  = 004020b9 in kmodloader[400000+3000]
[   18.606387] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
[   18.613598] epc = 77e2fb34 in libc.so[77dbc000+ab000]
[   18.618294] ra  = 004020b9 in kmodloader[400000+3000]
[   18.658131] Registering RTL8306SDM switch with Chip ID: 0x5988, version: 0x0000
[   18.664048] Realtek RTL8306S 1e180000.etop-ffffffff:00: attached PHY driver (mii_bus:phy_addr=1e180000.etop-ffffffff:00, irq=POLL)
[   18.689145] NET: Registered PF_INET6 protocol family
[   18.732680] Segment Routing with IPv6
[   18.735075] In-situ OAM (IOAM) with IPv6
[   18.739006] NET: Registered PF_PACKET protocol family
[   18.743974] 8021q: 802.1Q VLAN Support v1.8
[   18.799588] Freeing unused kernel image (initmem) memory: 5376K
[   18.804004] This architecture does not have kernel memory protection.
[   18.810358] Run /init as init process
[   20.969781] init: Console is alive
[   20.972722] init: - watchdog -
[   21.024556] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[   21.062970] dwc2 1e101000.usb: supply vusb_d not found, using dummy regulator
[   21.069356] dwc2 1e101000.usb: supply vusb_a not found, using dummy regulator
[   21.118943] dwc2 1e101000.usb: DWC OTG Controller
[   21.122412] dwc2 1e101000.usb: new USB bus registered, assigned bus number 1
[   21.129059] dwc2 1e101000.usb: irq 62, io mem 0x1e101000
[   21.137204] hub 1-0:1.0: USB hub found
[   21.141466] hub 1-0:1.0: 1 port detected
[   21.155934] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[   21.165935] init: - preinit -
[   21.269863] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[   21.275569] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[   21.499647] usb 1-1: new high-speed USB device number 2 using dwc2
[   21.504571] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[   21.511475] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[   21.518354] dwc2 1e101000.usb: Mode Mismatch Interrupt: currently in Host mode
[   21.653768] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   21.693278] random: procd: uninitialized urandom read (4 bytes read)
[   21.771791] hub 1-1:1.0: USB hub found
[   21.774423] hub 1-1:1.0: 4 ports detected
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[   22.859647] random: crng init done
[   24.139326] procd: - early -
[   24.142066] procd: - watchdog -
[   25.026977] procd: - watchdog -
[   25.031002] procd: - ubus -
[   25.118165] procd: - init -
Please press Enter to activate this console.
[   27.835906] kmodloader: loading kernel modules from /etc/modules.d/*
[   29.039352] IFXOS, Version 1.7.1 (c) Copyright 2009, Lantiq Deutschland GmbH
[   29.089168] Lantiq TAPI device driver, version 3.13.0.2, (c) 2001-2010 Lantiq Deutschland GmbH
[   29.129747] NET: Registered PF_ATMPVC protocol family
[   29.133355] NET: Registered PF_ATMSVC protocol family
[   29.186075] PPP generic driver version 2.4.2
[   29.222932] IFX MEI Version 5.00.00
[   29.280913] Infineon CPE API Driver version: DSL CPE API V3.24.4.4
[   29.338734] Loading modules backported from Linux version v6.5-0-g2dde18cd1d8f
[   29.344644] Backport generated by backports.git v5.15.92-1-56-g5fb2ccb6b9e8
[   29.410651] Lantiq MIPS24KEc MPS driver, version 2.2.2.5, (c) 2006-2010 Lantiq Deutschland GmbH
[   29.417809] request_timer(3, 0x000001AE, 1)...
[   29.417911] successful!
[   29.425782] Lantiq VMMC device driver, version 1.9.0.3, (c) 2006-2010 Lantiq Deutschland GmbH
[   29.532066] NET: Registered PF_PPPOX protocol family
[   30.200637] kmodloader: done loading kernel modules from /etc/modules.d/*
[   35.301138] urngd: v1.0.2 started.



BusyBox v1.36.1 (2024-01-21 18:18:06 UTC) built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt SNAPSHOT, r24879-abbf28637e
 -----------------------------------------------------
=== WARNING! =====================================
There is no root password defined on this device!
Use the "passwd" command to set up a new password
in order to prevent unauthorized SSH logins.
--------------------------------------------------
~~~

abajk commented:
I forgot to add that my device is broken. A BGA ball has probably cracked under the CPU and without pressing it down with a  mug it won't even start booting u-boot. I don't know how much this could be related to the errors that appear in dmesg.

Ansuel commented:
@abajk did you consider the over trick to reflow the entire pcb? might have good result with some kepton tape on the right place.

Ansuel commented:
Anyway yes would love some testing by know working router since it may be a problem with some pin on ram....

Ansuel commented:
@Leo-PL can you by chance test or do you have user to request some testing?

Leo-PL commented:
I can test on Easybox 802 in the coming days. Hopefully I won't need to disassemble my device

DragonBluep commented:
> @Ansuel I can test on Easybox 802 in the coming days. Hopefully I won't need to disassemble my device once more to unbrick. @abajk maybe some folks in HSWaw can help you out fixing the device.
> 
> ```
> [ 18.205794] do_page_fault(): sending SIGSEGV to modprobe for invalid read access from 0000004c
> [ 18.212988] epc = 77e82b34 in libc.so[77e0f000+ab000]
> [ 18.217699] ra = 004020b9 in kmodloader[400000+3000
> ```
> 
> This looks like another unchecked NULL pointer to struct dereference somewhere. Very unlikely to be caused by hardware, since it repeats and is very consistent.

This issue seems to be introduced by https://github.com/openwrt/openwrt/commit/ced3fbcda18e200315cb077f7e2b362ae52ee065. I can catch the same log on the ath79 target.

Leo-PL commented:
@Ansuel works on my device. Strangely I can't see the segfaults in the kernel log, but I built a bare minimum image, maybe there's no kernel module to load with parameters or whatnot.

Ansuel commented:
The image is with todays commits? I pushed some fix to ubox so maybe I indirectly fix the extra warning. Wonder if @DragonBluep can confirm.

Leo-PL commented:
No, direct clone from @abajk's repo.

Ansuel commented:
@Leo-PL very strange... @DragonBluep  anyway can create an issue so we can track it there? (and tag me)

Ansuel commented:
@abajk can you consider pushing this upstream? They should accept it right away...

abajk commented:
I sent the [patch upstream](https://patchwork.kernel.org/project/linux-mips/patch/20240122184709.9625-1-olek2@wp.pl/). I expect it will take some time until it gets merged because the mips mailing list is not very active.
